react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -94,6 +90,8 @@ export class QRCode extends Component {
       .then(this.props.history.push('/visitor/end'))
       .catch(error => {
         console.log('ERROR HAPPENING AT FETCH /postActivity', error);
+        this.props.history.push('/visitor/login');
+        throw error;
```

Comment:
If you are sending the user to another route (`/visitor/login/` or `/visitor/qrerror` or anything), you shouldn't also need to `throw` here, you are handling your error by redirecting the user, so the throw is unnecessary as far as I can see.